source/auth/auth.rst

```diff
@@ -1251,172 +1251,527 @@ mechanism
 
 mechanism_properties
     PROVIDER_NAME
-        Drivers MUST allow the user to specify a name for using a service
-        to obtain credentials that is one of ["aws"].
-    REQUEST_TOKEN_CALLBACK
-        Drivers MUST allow the user to specify a callback of the form
-        "onRequest" (defined below), if the driver supports
-        providing objects as mechanism property values.  Otherwise the driver MUST allow it as a MongoClientOption.
-    REFRESH_TOKEN_CALLBACK
-        Drivers MUST allow the user to specify a callback of the form
-        "onRefresh" (defined below), if the driver supports
-        providing objects as mechanism property values.  Otherwise the driver MUST allow it as a MongoClientOption.
+        The name of a built-in OIDC provider integration to use to obtain
+        credentials. The value MUST be one of ["aws"]. If both ``PROVIDER_NAME``
+        and an `OIDC Callback`_ are provided for the same ``MongoClient``
+        (either via the ``CALLBACK`` mechanism property or a ``MongoClient``
+        configuration), the driver MUST raise an error.
+
+    CALLBACK
+        An `OIDC Callback`_ that returns credentials for OIDC providers that do
+        not have a built-in integraiton. Drivers MAY allow the user to specify
+        an `OIDC Callback`_ using a ``MongoClient`` configuration instead of a
+        mechanism property, depending on what is conventional for the driver.
+        Drivers MUST NOT support both the ``CALLBACK`` mechanism property and
+        the ``MongoClient`` configuration.
+
+    CALLBACK_TYPE
+        The type of `OIDC Callback`_. The value MUST be one of ["machine",
+        "human"]. Drivers MUST NOT allow the user to specify ``CALLBACK_TYPE``
+        in the connection string. Drivers MAY allow the user to specify the
+        callback type using a ``MongoClient`` configuration instead of a
+        mechanism property to be consistent with how the `OIDC Callback`_ is
+        configured. If an `OIDC Callback`_ is configured and ``CALLBACK_TYPE``
+        is not specified, the driver MUST raise an error. This property is only
+        required for drivers that support the `Human Authentication Flow`_.
```

Comment:
I think we have to say that the property is required in all cases.  Otherwise introduction of the human flow in a subsequent release will be a breaking change, because an existing app that has registered a machine callback will error on upgrade.

Reply:
Good point, that would create a breaking change for drivers that add human auth flow support later. Based on that and your other comments, I think we should not use `CALLBACK_TYPE`. Instead, we should replace it with either a property like `USE_HUMAN_CALLBACK:true` or a property like `HUMAN_CALLBACK` that expects a callback function and is mutually exclusive with `CALLBACK`.

Reply:
Resolved by adding OIDC_HUMAN_CALLBACK